azure-pipelines.perf-test.yml

```diff
@@ -10,86 +10,132 @@ variables:
-    inputs:
-      SourcePath: 'packages/fluentui/perf-test-northstar/dist'
-      azureSubscription: $(azureSubscription)
-      storage: $(azureStorage)
-      ContainerName: '$web'
-      BlobPrefix: '$(deployBasePath)/perf-test-northstar'
-
-  - task: AzureUpload@2
-    displayName: Upload Perf Test Result to PR deploy site (Fluent)
-    inputs:
-      SourcePath: 'apps/perf-test/dist'
-      azureSubscription: $(azureSubscription)
-      storage: $(azureStorage)
-      ContainerName: '$web'
-      BlobPrefix: '$(deployBasePath)/perf-test'
-
-  - task: GithubPRComment@0
-    displayName: 'Post Perf Results to Github Pull Request'
-    inputs:
-      githubOwner: microsoft
-      githubRepo: 'fluentui'
-      blobFilePath: '$(Build.SourcesDirectory)/$(PerfCommentFilePath)'
-      status: '$(PerfCommentStatus)'
-      uniqueId: 'perfComment9423'
-
-  - script: |
-      yarn stats:build
-    condition: eq(variables.isPR, false)
-    displayName: Bundle Statistics (master only)
-
-  - script: |
-      yarn perf
-    condition: eq(variables.isPR, false)
-    displayName: Performance Tests (master only)
-
-  # HEADS UP: also see tag-version-prefix in fluentui-publish.js
-  - script: |
-      yarn stats:save --tag=`git tag --points-at HEAD | grep ^@fluentui/react-northstar_v | grep -o 'northstar_v.*'`
-    condition: eq(variables.isPR, false)
-    displayName: Save Statistics to DB (master only)
-    env:
-      STATS_URI: $(STATS_URI)
-
-  - template: .devops/templates/cleanup.yml
+jobs:
+  - job: PerfTestNorthstar
+    displayName: Perf test for @fluentui/react-northstar
+    workspace:
+      clean: all
+    steps:
+      - template: .devops/templates/tools.yml
+
+      - task: Bash@3
+        inputs:
+          filePath: yarn-ci.sh
+        displayName: yarn
+
+      - script: |
+          yarn just check-if-package-affected --packageAffected @fluentui/react-northstar
+        workingDirectory: packages/fluentui/perf-test-northstar
+        displayName: Check if @fluentui/react-northstar was modified
+
+      # Build @fluentui/perf-test-northstar
+      - script: |
+          yarn lage build --to @fluentui/perf-test-northstar --to @fluentui/react-conformance --verbose
+        condition: and(variables.PackageAffected, variables.isPR)
+        displayName: Build to Perf Test (Northstar)
+
+      # @fluentui/perf-test-northstar Baseline
+      - script: |
+          yarn perf:test:base
+        condition: and(variables.PackageAffected, eq(variables.isPR, false))
+        workingDirectory: packages/fluentui/perf-test-northstar
+        displayName: Run Perf Test Base (Northstar)
+
+      # Run @fluentui/perf-test-northstar
+      - script: |
+          yarn perf:test
+        condition: and(variables.PackageAffected, variables.isPR)
+        workingDirectory: packages/fluentui/perf-test-northstar
+        displayName: Run Perf Test (Northstar)
+
+      - task: AzureUpload@2
+        condition: and(variables.PackageAffected, variables.isPR)
```

Comment:
For all these checks you may need explicit `eq(variables.isPR, true)`--IIRC I'd written some checks the way you have them now in the past, and @ling1726 or somebody noticed that it wasn't working because variables in this context are actually strings (always truthy), or something along those lines.

Reply:
In this case, `PackageAffected` is either `true` or `null` so the condition does fail when it's `null`